indexdata.go

```diff
@@ -166,6 +166,14 @@ func (d *indexData) getChecksum(idx uint32) []byte {
 	return d.checksums[start : start+crc64.Size]
 }
 
+func (d *indexData) getLanguage(idx uint32) uint16 {
+	if len(d.languageMap) > 256 {
+		return uint16(d.languages[idx*2]) |
+			uint16(d.languages[idx*2+1])<<8
```

Comment:
Is it worth the memory savings of N file bytes to use this more complicated mapping vs just always doing `languages []uint16`?

If you do want to keep this, maybe there is a nicer way to represent this? eg pulling languages and languageMap out into its own type that you query, rather than making this logic that is coupled in both indexData and indexBuilder. This is not a blocker, just more something to think about.
If you do want to keep this space saving, can you somehow abstract `[]byte` into something which wraps `[]byte` to make it clearer how this all fits together. Right now

Reply:
That breaks backwards compatibility in a slightly nasty way, even though the language mappings are different. I can wrap this in a better datastructure.